web/src/components/FileUpload.jsx

```diff
@@ -48,32 +66,28 @@ export default function FileUpload() {
     setDefaultFileList(fileList);
   };
 
-  const handleCancel = () => {
-    setPreviewVisible(false);
-  };
-
   const handlePreview = async file => {
-    setPreviewVisible(true);
     getBase64(file.originFileObj).then(f => setPreviewImage(f));
+    /**  If you call getBase64(file) you get parameter must be 'blob'. 
+     * If you try to store it with [getBase64(file.originFileObj)] you get the same result on line 62
```

Comment:
What do you mean by line 62?

Reply:
Forgot the update the line number after deleting some un-used hooks. Its the line 
`// [(getBase64(file)).then(f => md5(f))]: data`  . (line 58 i think, at the end of annotateImage call). 
<img width="690" alt="Screen Shot 2020-12-25 at 22 57 18" src="https://user-images.githubusercontent.com/40474912/103141381-9890cd80-4704-11eb-81cb-458e6b13fe35.png">

Could not use this line as a key; it is possible to insert the annotation to the dictionary with this key except that I came-across the "Parameter must be 'blob' " exception when i tried to access the element.

Reply:
The then part looks like a promise. are you sure it is not async?

Reply:
Re-Designed the dictionaries, thus im resolving this conversation. You can comment on the one about the renaming the file.